docs/proposals/pod-troubleshooting.md

```diff
@@ -0,0 +1,489 @@
+the entrypoint of a pod that's crash looping:
+
+```
+kubectl debug --copy-of=target-pod --pod=target-pod-copy -it --attach --container=crashing-container --command -- sh
+```
+
+The `--container` and `--attach` arguments would follow established conventions
+for choosing a default container (kubectl exec) and attaching if stdin was
+specified (kubectl run) and so are redundant in the above example. When
+`--copy-of` is provided, we can construct a reasonable default for `--pod`(e.g.
+`${copy-of}-copy`), making the minimal form of this command:
+
+```
+kubectl debug --copy-of=target-pod -it --command -- sh
+```
+
+One can specify a container that doesn't currently exist in the pod spec to
+create a new one:
+
+```
+kubectl debug --copy-of=target-pod -it -c shell --image=debian
+```
+
+Command line arguments allow simple, single-container changes but are
+insufficient for tasks like modifying volumes and their mount points. For more
+complex changes, we can provide an `--edit` workflow to adjust the generated pod
+spec before creation:
+
+```
+# Similar to the following workflow:
+#   kubectl debug --dry-run -o yaml ... > $temp_file
+#   $EDITOR $temp_file
+#   kubectl create -f $temp_file
+kubectl debug --copy-of=target-pod -it -c shell --image=debian --edit
+```
+
+#### Scheduling conflicts
+
+Attempting to duplicate some resources will cause the resulting pod to be
+unschedulable, for example creating a second read-write volume mount of a
+gcePersistentDisk will create a disk conflict. This is the same problem faced by
+scaling a deployment, so one way to provide a better user experience is to
+validate the generated config similar to `ValidatePodTemplateSpecForRC()` with
+replicas > 1 as `kubectl scale` does.
+
+Validating client side is not optimal, and currently only gcePersistentDisk
+provides a check in `ValidatePodTemplateSpecForRC()`. Since *Running Debug* must
+be implemented server-side, a better long term strategy may be to migrate *Copy
+Debug* server-side as well to take advantage of easier validation against
+current cluster state.
+
+#### Stripping labels
+
+Only the spec will be copied from the running pod. Copying metadata such as
+labels might result in the pod copy receiving traffic or being killed
+immediately by a replication controller. It would be trivial to provide a
+`--copy-labels` option if this is desired behavior.
+
+### Running Debug Mode
+
+*Running Debug* mode is a `kubectl debug` invocation that results in a new
+container being introduced in the running pod. This *debug container* is not
+part of the pod spec, which remains immutable, and is not restarted according to
+the pod's `restartPolicy`.
+
+The status of a *debug container* is reported in a new `DebugContainerStatuses`
+field of `PodStatus`, which is a read-only list of `ContainerStatus` that
+contains an entry for every *debug container* that has ever run in this pod.
+Additionally, a new boolean `Tainted` in `PodCondition` changes from false to
```

Comment:
This is sort of at the edge of status's intended meaning.  The litmust test was always "if status were lost, could it be recreated by inspection" ?  I guess it could in this case, if the runtime implementations are careful to squirrel away data in tings like docker labels